File: frontend/config_loader.js
const fs = require('fs');
const path = require('path');

function loadConfig() {
    const configPath = path.join(__dirname, '..', 'config', 'frontend_config.json');
    
    try {
        const configData = fs.readFileSync(configPath, 'utf8');
        return JSON.parse(configData);
    } catch (error) {
        console.error('⚠️ Error loading configuration:', error.message);
        console.log('⚠️ Using default configuration');
        return {
            server: {
                port: 8080,
                host: '0.0.0.0'
            },
            services: {
                translation: {
                    host: 'localhost',
                    port: 50051
                },
                chat: {
                    host: 'localhost',
                    port: 50052
                },
                auth: {
                    host: 'localhost',
                    port: 50053
                },
                tts: {
                    host: 'localhost',
                    port: 50054
                }
            }
        };
    }
}

module.exports = { loadConfig }; 

File: backend/config_loader.py
import json
import os

def load_config(service_name):
    """
    Load configuration for a specific service from the config file.
    
    Args:
        service_name (str): Name of the service (translation, chat, auth, tts)
        
    Returns:
        dict: Configuration for the specified service
    """
    config_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'config', 'backend_config.json')
    
    try:
        with open(config_path, 'r') as f:
            config = json.load(f)
            return config['services'][service_name]
    except FileNotFoundError:
        print(f"⚠️ Configuration file not found at {config_path}")
        return None
    except KeyError:
        print(f"⚠️ Configuration for service '{service_name}' not found")
        return None
    except json.JSONDecodeError:
        print(f"⚠️ Invalid JSON in configuration file {config_path}")
        return None 